tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -153,125 +153,93 @@ isCompilerGenerated(const string data)
 }
 
 /**
- * Parses a line of data and returns the declaration type of the data string, using regex expressions. 
+ * Parses a line of data and returns the declaration type of the data string. 
  *
  * @param[in]  data: the string ot be parsed.
  * @param[out] tag                   : the type of the declaration of the data string
- * @param[out] parsedDeclarationData : the declaration data split into name, type and members (if applicable)
+ * @param[out] typeID                : the ID of the entity being declared
  */
 static Dwarf_Half
-parseDeclarationLine(const string data, Parsed_Declaration_Data *parsedData)
-{	
+parseDeclarationLine(const string data, double &typeID)
+{
 	Dwarf_Half ret = DW_TAG_unknown;
-	if (!data.empty()) {
-		regex typeDefExpression("(^):t"+INTEGER+"="+"(c|d|e|g|D|k|m|w|M|\\*|\\&|V|Z|f|a)?(.*)");
-		regex constantExpression("(^)"+NAME+":c="+"(b|c|i|r|s|C|S)?;(.*)");
-		regex structureExpression("=s"+NUM_BYTES);
-		regex unionExpression("=u"+NUM_BYTES);
-		regex classExpression("=Y"+NUM_BYTES+"(?:"+CLASS_KEY+")+(.*)(\\()");
-		regex classStructure("Y"+NUM_BYTES+"s(.*)");
-		regex classUnion("Y"+NUM_BYTES+"u(.*)");
-		regex enumExpression("=e"+INTEGER);
-
-		smatch m;
-		if (regex_search(data, m, typeDefExpression)) {
-			ret = DW_TAG_typedef;
-			
-			/* Split the string in half at the '=' to get name, type halves */
-			str_vect tmp = split(data, '=');
-			
-			/* Split the half containing the name into another 2 halves, to get the name and typeID */
-			str_vect tmp2 = split(tmp[0], ':');
-			parsedData->name = tmp2[0];
-			parsedData->typeID = tmp2[1];
-			parsedData->type = tmp[1];
-		} else if (regex_search(data, m, constantExpression)) {
+	
+	/* split the declaration into the name and the information about the declaration */
+	size_t indexOfFirstColon = data.find_first_of(':');
```

Comment:
I'll add code to check that `data` contains a colon, and add documentation for the parameters being passed in.